app/src/components/pages/collectionsPage/collectionsPage.tsx

```diff
@@ -1,17 +1,113 @@
 "use client";
 import PageLayout from "../../pageLayout/pageLayout";
-import React from "react";
-import SearchResults from "../../search/results";
+import React, { useState, useRef, useEffect } from "react";
 import {
   Box,
   Heading,
   HorizontalRule,
   SearchBar,
+  Menu,
+  Text,
+  Pagination,
 } from "@nypl/design-system-react-components";
+import { useSearchParams, usePathname, useRouter } from "next/navigation";
 import { headerBreakpoints } from "../../../utils/breakpoints";
+import { CardsGrid } from "../../grids/cardsGrid";
+import LaneLoading from "../../lane/laneLoading";
+import { displayResults, totalNumPages } from "../../../utils/utils";
+import type { SyntheticEvent } from "react";
 import { createAdobeAnalyticsPageName } from "@/src/utils/utils";
 
 export const CollectionsPage = ({ data }) => {
+  const { push } = useRouter();
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  const collections = data.collection; // typeof data.collection === "object" ? [data.collection] : data.collection // this breaks empty search
+  const headingRef = useRef<HTMLHeadingElement>(null);
+
+  // pagination
+  const numFound = data.numFound ? data.numFound : data.numResults;
+  const totalPages = totalNumPages(numFound, data.perPage);
+
+  // defaults
+  let currentPage = Number(searchParams.get("page")) || 1;
+  let currentSort = searchParams.get("sort") || "date-desc";
+  let currentCollectionKeyword = searchParams.get("collection_keyword") || "";
+
+  const handleSearchSubmit = async (e: SyntheticEvent) => {
+    e.preventDefault();
+    const queryString = createQueryStringFromObject({
+      collection_keyword: currentCollectionKeyword,
+      sort: currentSort,
+      page: currentPage,
+    });
+    setIsLoaded(false);
+    await push(`${pathname}?${queryString}`);
+    setTimeout(() => {
+      setIsLoaded(true);
+      headingRef.current?.focus();
+    }, 2000);
+  };
+
+  // I'm not actually sure if this is necessary but it's in the research catalog
+  const handleSearchChange = (e: SyntheticEvent) => {
+    const target = e.target as HTMLInputElement;
+    currentCollectionKeyword = target.value;
+  };
```

Comment:
Yes. This works but change it to use state instead. This means you need to use `searchParams.collection_keyword` as the default state value such as `const [currentCollectionKeyword, setCurrentCollectionKeyword] = useState<string>(searchParams.collection_keyword || ");`

Then this function can use the `setCurrentCollectionKeyword` function to update the value.
You should not update a variable directly in this manner in order to reduce rerendering inconsistencies.

Reply:
do you think we only need to use state for the input field/currentCollectionKeyword? I feel like the other two (sort,pagination) can be managed the way that they are right now.

Reply:
I'm going to leave the approach up to you, but I hope this helps:

- Search for "cat"
- click on the "clear" X button in the input
- perform an empty search

The single "cat" result still shows. When you manually remove the text from the input and perform an empty search, it works. You have to somehow keep track of the keyword for this use case.